### PR TITLE
ci: addition of commit message linter

### PR DESCRIPTION
**Pull request by tiborsimko**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,7 +42,7 @@ jobs:
       - name: Runs shell script static analysis
         run: |
           sudo apt-get install shellcheck
-          ./run-tests.sh --check-shellscript
+          ./run-tests.sh --check-shellcheck
```
Comment (mdonadoni) [inline .github/workflows/ci.yml:45]: There are some inconsistencies here and in `run-tests.sh`, as sometimes we call `--check-shellcheck` and some other times `--check-shellscript`. For example, in `run_all` the function `check_shellcheck` is called, but there is no function with that name.

PS: should we maybe add also a default case in `run-tests.sh` to catch typos when calling `./run-tests.sh --check-...`? Something like this:
```bash
for arg in "$@"
do
    case $arg in
        [...]
        *) echo "invalid argument $arg" && exit 2;;
    esac
done
```

File: run-tests.sh [hunk 2]
```diff
@@ -58,7 +61,8 @@ fi
 for arg in "$@"
 do
     case $arg in
-        --check-shellscript) check_script;;
+        --check-commitlint) check_commitlint "$@";;
```
Comment (mdonadoni) [inline run-tests.sh:64]: The passing of `$@` to the function breaks when multiple operation are requested, like this:
```
./run-tests.sh --check-pytest --check-commitlint
```

Maybe we can remove the for loops that iterates over all the arguments? I personally never pass multiple checks at the same time, so maybe also others do not do that?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tiborsimko) [reply .github/workflows/ci.yml:45]: Wanted to change the command everywhere to have the tool name <-> option name consistency, but apparently forgot some parts. Fixed.

Good idea to do catch-all rule to return error. Added.

File: run-tests.sh (hunk 2 above)
Comment (tiborsimko) [reply run-tests.sh:64]: Yes, I'm doing the same as you, so I agree. Changed.

**Review by mdonadoni — APPROVED** [review]
LGTM!

**Pull request merged**